File: test/end-to-end/PointClass/Ruby/named_point.rb
require_relative "./point"

class NamedPoint < Point
    attr_accessor :name

    def initialize(x, y, name)
        super(x, y)
        self.name = name
    end

    def get_label()
        return "%s at [%d, %d] with vector %d" % [self.name, self.x, self.y, self.vector]
    end
end

File: test/end-to-end/PointClass/PHP/named_point.php
require_relative "./point"

<?php

class NamedPoint < Point
    attr_accessor :name;

    function __constructor(x, y, name)
        parent::(x, y);
        self.name = name;
    end

    def get_label()
        return '' . self.name . ' at [' . self.x . ', ' . self.y . '] with vector ' . self.vector . '';
    end
end
